### PR TITLE
Add appointment management feature with serialization, validation, and testing

### PR DESCRIPTION
**Pull request by pradyuprasad**
### Key Changes:

1. **Appointment Model:**
   - Introduced the `Appointment` class to represent patient appointments.
   - Added validation for appointment times (start time must be before end time) with custom `InvalidAppointmentException`.
   - Implemented methods for marking an appointment as completed and comparing appointments through the `equals()` method.

2. **AppointmentManager:**
   - Implemented the `AppointmentManager` as a singleton to manage appointments across the application.
   - Added methods to add, remove, and retrieve appointments while ensuring conflict detection for overlapping appointments.
   - Integrated `AppointmentManager` into `ModelManager` to handle the addition and removal of appointments.

3. **Serialization and Deserialization:**
   - Created `JsonAdaptedAppointment` for Jackson serialization and deserialization of appointments.
   - Updated `JsonAdaptedPerson` to include appointments during JSON conversion.
   - Ensured appointments are properly serialized and deserialized along with persons in the application.

4. **Person Class:**
   - Updated the `Person` class to include appointment handling:
     - Added methods to add, remove, and retrieve appointments.
     - Ensured that appointments are immutable once added.
   - Updated the `PersonBuilder` class to allow setting appointments when constructing a `Person`.

5. **Testing:**
   - Added unit tests for the `Appointment` class to cover validation, appointment creation, and comparison.
   - Added tests for `JsonAdaptedAppointment` and `JsonAdaptedPerson` to ensure proper serialization and deserialization of appointments.
   - Updated test utilities (`TypicalPersons`, `PersonBuilder`) to include appointments in test data.

### Additional Details:
- Conflict detection ensures that no two appointments overlap in time.
- Appointment times are strictly validated, and invalid inputs (e.g., end time before start time) are rejected with appropriate exceptions.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -213,6 +219,42 @@ public boolean isSamePerson(Person otherPerson) {
...
+    /**
+     * Adds an appointment to the list of appointments.
+     *
+     * @param appointment The appointment to be added.
+     */
+    public void addAppointment(Appointment appointment) {
+        appointments.add(appointment);
+    }
+
+    /**
```
Comment (AngPengXuan) [inline src/main/java/seedu/address/model/person/Person.java:240]: Good job adding clear comments to all the new methods thus far, remember to add the corresponding test cases!

File: src/main/java/seedu/address/model/person/Person.java [hunk 2]
```diff
@@ -35,10 +36,15 @@ public class Person {
     private final Set<Nric> caregivers = new HashSet<>();
     private final Set<Nric> patients = new HashSet<>();
 
+    // Appointments
+    private final Set<Appointment> appointments = new HashSet<>();
```
Comment (AngPengXuan) [inline src/main/java/seedu/address/model/person/Person.java:40]: The comment Appointments/Notes does't seem very insightful? Its the same as the variable name.

File: src/main/java/seedu/address/model/Model.java [hunk 3]
```diff
@@ -93,4 +95,14 @@ public interface Model {
     void deleteLink(Person patient, Person caregiver);
 
     Person getPerson(Nric nric);
+
+    ObservableList<Person> getUnfilteredPersonList();
```
Comment (AngPengXuan) [inline src/main/java/seedu/address/model/Model.java:99]: Good use of the Model interface to include appointments!

File: src/main/java/seedu/address/model/person/Person.java (hunk 2 above)
Comment (pradyuprasad) [reply src/main/java/seedu/address/model/person/Person.java:40]: noted, am removing

**Review by lavanyagarg112 — APPROVED** [review]
LGTM

**Pull request merged**